supabase/models/messaging/newConvoStart.ts

```diff
@@ -8,13 +8,40 @@ export default async function newConvoStart(
   if (userID && donorID && itemID) {
     try {
       const supabase = newClient();
-      const { data: conversationData } = await supabase
+      /**
+       * @description before creating a new conversation, we check if such a conversation already exists in the database,
+       * if we find such a conversation, we return its ID that we can use further in the code,
+       * if such a conversation does not exist in the database, then we create a new conversation
+       */
+
+      // Checking if there is already a conversation between these users about this product
+      const { data: existingConversations, error } = await supabase
+        .from('user_conversations')
+        .select('conversation_id')
+        .eq('user_id', userID)
+        .eq('partner_id', donorID)
+        .eq('item_id', itemID);
+      if (error) {
+        console.error('Error checking for existing conversations:', error);
+        return null;
+      }
+      // If the conversation already exists, we return its ID
+      if (existingConversations.length > 0) {
+        console.log('such a conversation already exists');
```

Comment:
Remove console.log

Reply:
Thanks for the comments, the changes have been made. I will be more careful in the future :)